nevergrad/optimization/recastlib.py

```diff
@@ -57,21 +71,54 @@ def _optimization_function(
         remaining: float = budget - weakself._num_ask
         while remaining > 0:  # try to restart if budget is not elapsed
             options: tp.Dict[str, tp.Any] = {} if weakself.budget is None else {"maxiter": remaining}
-            res = scipyoptimize.minimize(
-                objective_function,
-                best_x if not weakself.random_restart else weakself._rng.normal(0.0, 1.0, weakself.dimension),
-                method=weakself.method,
-                options=options,
-                tol=0,
-            )
-            if res.fun < best_res:
-                best_res = res.fun
-                best_x = res.x
+            # options: tp.Dict[str, tp.Any] = {} if self.budget is None else {"maxiter": remaining}
+            if weakself.method == "CmaFmin2":
+                def cma_objective_function(data):
+                    # Hopefully the line below does nothing if unbounded and rescales from [0, 1] if bounded.
+                    if self._normalizer is not None:
+                        data = self._normalizer.backward(np.asarray(data, dtype=np.float32))
+                    return objective_function(data)
+
+                # cma.fmin2(objective_function, [0.0] * self.dimension, [1.0] * self.dimension, remaining)
+                x0 = 0.5 * np.ones(self.dimension)
+                num_calls = 0
+                while budget - num_calls > 0:
+                    options = {"maxfevals": budget - num_calls, "verbose": -9}
+                    if p.helpers.Normalizer(self.parametrization).fully_bounded:
```

Comment:
Instantiating a normalizer is slow, do it only in init
```suggestion
                    if self._normalizer is not None:
```